Kom4 12 search bar function, filter, validacija

AGiedrius opened the pull request:
Backend
Paieškos funkcija pagal fragmenta;
Filtravimo funkcionalumas pagal
    type, 
    preparation_time,
    servings,;
Pvz.: http://localhost:3001/api/v1/recipes/?q=pica&type=non-veg
 http://localhost:3001/api/v1/recipes/?type=veg ir t.t.

Validatorius

FlorijanDem committed:
Merge pull request #27 from FlorijanDem/develop

Push this week progress

AGiedrius committed:
Add search functionality with similarity search and type filtering

AGiedrius committed:
Merge search functionality with similarity search and type filtering

AGiedrius committed:
KOM4-12: Implemented search bar functionality with filters and validators

AGiedrius committed:
Merge branch 'main' into KOM4-12/Search_bar_function/filter_function/validator

FlorijanDem commented on backend/app.js:
Instead of writing a new URL every time, you can simple use "origin: true, // Reflects the request origin"

AGiedrius committed:
Code formatting and comments cleanup

AGiedrius committed:
Merge branch 'develop' of https://github.com/FlorijanDem/receptoKnyga into develop

AGiedrius committed:
Merge branch 'develop' into KOM4-12-search-bar-function

AGiedrius committed:
Save local changes in search bar function

AGiedrius committed:
feat: add filter by product functionality to recipe search

- Added product parameter handling in controller
- Updated search query in model to filter by product
- Updated API documentation in README
- Added product parameter validation
